sonic-pcied/scripts/pcied

```diff
@@ -151,6 +160,31 @@ class DaemonPcied(daemon_base.DaemonBase):
 
         self.status_table.set("status", fvs)
 
+    # Check if any interface is in detaching mode by querying the state_db
+    def is_dpu_in_detaching_mode(self, pcie_dev):
+        try:
+            # Ensure detach_info is not None
+            if self.detach_info is None:
+                self.log_debug("detach_info is None")
+                return False
+
+            # Query the state_db for the device detaching status
+            detach_info_keys = list(self.detach_info.getKeys())
+            if not detach_info_keys:
+                return False
+
+            for key in detach_info_keys:
+                dpu_info = self.detach_info.get(key)
+                if dpu_info:
+                    bus_info = dpu_info.get(PCIE_DETACH_BUS_INFO_FIELD)
+                    dpu_state = dpu_info.get(PCIE_DETACH_DPU_STATE_FIELD)
+                    if bus_info == pcie_dev and dpu_state == "detaching":
```

Comment:
@vvolam I am not sure whether this "detaching" state is defined in the HLD. Can you point me to the HLD?

Can we make use of existing DPU state? What will be the DPU state once the DPU is detached?

Reply:
@prgeor Here is the schema defined in the HLD: https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/reboot/reboot-hld.md#pcie_detach_info-schema-in-statedb

Reply:
@prgeor Since we want to differentiate between intentional removal of DPU, HLD has defined the above schema. If we rely on DPU state, we may not be able to differentiate between intentional and error states.